### PR TITLE
feat: Adds number validation for WhatsApp inbox at the creation step

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

Fixes https://github.com/chatwoot/product/issues/677

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## How Has This Been Tested?
https://www.loom.com/share/ab3dbeaba4484c7d9954ac9f413441fa

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by pranavrajs — APPROVED** [review]
Code looks fine. @iamsivin Can you please test it with actual values?

**Review by nithindavid — CHANGES_REQUESTED** [review]
image
The error message for numbers with spaces is misleading, can you add correct the text?

**Comment by iamsivin** [comment]
> Code looks fine. @iamsivin Can you please test it with actual values?

@pranavrajs Tested it with some dummy values. LGTM.

File: app/javascript/dashboard/i18n/locale/en/inboxMgmt.json [hunk 1]
```diff
@@ -214,7 +214,7 @@
         "PHONE_NUMBER": {
           "LABEL": "Phone number",
           "PLACEHOLDER": "Please enter the phone number from which message will be sent.",
-          "ERROR": "Please enter a valid value. Phone number should start with `+` sign."
+          "ERROR": "Please enter a valid value. Phone number should start with a `+` sign and not contain any spaces."
```
Comment (nithindavid) [inline app/javascript/dashboard/i18n/locale/en/inboxMgmt.json:217]:
```suggestion
         "ERROR": "Please provide a valid phone number that starts with a `+` sign and does not contain any spaces."
```

**Review by nithindavid — APPROVED** [review]

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**